src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -515,5 +517,29 @@ public override void Shutdown()
             Stop();
             HostingEnvironment.InitiateShutdown();
         }
+
+        public async Task DelayUntilHostReady(int timeoutSeconds = WebScriptHostHandler.HostTimeoutSeconds, int pollingIntervalMilliseconds = WebScriptHostHandler.HostPollingIntervalMilliseconds, bool throwOnFailure = true)
```

Comment:
I think we're saying the same thing. If you have 1000 RPS coming in, and you start delaying requests, they'll start queuing up somewhere (or getting canceled) -- we're letting IIS/Web API handle it. As opposed to, say, using our own request-queueing mechanism like we do elsewhere.

I like it -- much simpler than what I thought it was going to be :-)